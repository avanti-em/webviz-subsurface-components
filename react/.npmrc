engine-strict=true
ignore-scripts=true
CYPRESS_INSTALL_BINARY=0