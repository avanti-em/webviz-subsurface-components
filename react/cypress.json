{
    "baseUrl": "http://localhost:6006/?path=/story/deckglmap--default",
    "video": true,
    "screenshotsFolder": "./cypress/snapshots/actual",
    "trashAssetsBeforeRuns": true,
    "env": {
        "failSilently": false,
        "ALWAYS_GENERATE_DIFF": false,
        "type": "base"
    }
}
