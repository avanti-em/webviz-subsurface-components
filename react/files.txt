cypress-visual-screenshots/comparison/defaultMap.cy-default-map-story.png
cypress-visual-screenshots/diff/defaultMap.cy-default-map-story.png
files.txt
