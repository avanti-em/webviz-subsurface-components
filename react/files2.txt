cypress-visual-screenshots/diff/defaultMap.cy-default-map-story.png
