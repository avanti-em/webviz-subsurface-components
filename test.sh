if [[ -d react/cypress/snapshots/diff ]]
then
    echo "::set-output name=docs_changed::true"
else
    echo "::set-output name=docs_changed::false"
fi