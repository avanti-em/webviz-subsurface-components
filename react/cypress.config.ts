import { defineConfig } from "cypress";
const getCompareSnapshotsPlugin = require("cypress-image-diff-js/dist/plugin");

export default defineConfig({
  component: {
    devServer: {
      framework: "create-react-app",
      bundler: "webpack",
    },
    video:false,
    setupNodeEvents(on, config) {
      getCompareSnapshotsPlugin(on, config);
    }
  },
});
