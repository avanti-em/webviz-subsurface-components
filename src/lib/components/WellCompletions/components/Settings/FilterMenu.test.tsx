import "@testing-library/jest-dom/extend-expect";
import { fireEvent, render, screen } from "@testing-library/react";
import "jest-styled-components";
import React from "react";
import { testStore, Wrapper } from "../../test/TestWrapper";
import FilterMenu from "./FilterMenu";

/*
The following tests are breaking due to jest trying to parse a non-js file,
needs investigation to solve this problem.
*/

describe("basic snapshot test Filter Menu", () => {
    it("click to open filter menu",  () => {
        fireEvent.click(screen.getByRole("button"));
        const { container } = render(
            Wrapper({ children: <FilterMenu /> })
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});