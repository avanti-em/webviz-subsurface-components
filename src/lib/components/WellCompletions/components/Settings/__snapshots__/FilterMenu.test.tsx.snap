// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Filter Menu snapshot test 1`] = `
.c2 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 > img:first-child,
.c2 > svg:first-child {
  margin-right: 8px;
}

.c2 > img:last-child,
.c2 > svg:last-child {
  margin-left: 8px;
}

.c2 > img:only-child,
.c2 > svg:only-child {
  margin-left: auto;
  margin-right: auto;
}

.c2 > span:first-child {
  margin-left: 32px;
}

.c2 > span:last-child {
  margin-right: 32px;
}

.c2 > span:only-child {
  margin-right: 0;
  margin-left: 0;
}

.c1 {
  margin: 0;
  padding: 0;
  background: transparent;
  height: 40px;
  width: 40px;
  padding-left: 0;
  padding-right: 0;
  border: 1px solid transparent;
  border-radius: 50%;
  margin: 0;
  color: var(--eds_interactive_primary__resting,rgba(0,112,121,1));
  font-family: Equinor;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 1.143em;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  cursor: pointer;
  display: inline-block;
}

.c1 svg {
  justify-self: center;
}

.c1::after {
  position: absolute;
  top: -6px;
  left: -4px;
  width: 48px;
  height: 48px;
  content: '';
}

.c1:hover {
  background: var(--eds_interactive_primary__hover_alt,rgba(222,237,238,1));
  border: 1px solid transparent;
  border-radius: 50%;
}

.c1:focus {
  outline: none;
}

.c1[data-focus-visible-added]:focus {
  outline: 1px dashed rgba(0,112,121,1);
  outline-offset: 4px;
}

.c1::-moz-focus-inner {
  border: 0;
}

.c1:disabled {
  cursor: not-allowed;
  background: transparent;
  border: 1px solid transparent;
  margin: 0;
  color: var(--eds_interactive__disabled__text,rgba(190,190,190,1));
  font-family: Equinor;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 1.143em;
  text-align: center;
}

.c1:disabled:hover {
  background: transparent;
}

.c1::before {
  position: absolute;
  top: 0;
  left: 0;
  width: auto;
  min-height: auto;
  content: '';
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div>
  <div
    class="c0"
  >
    <div>
      <button
        class="c1"
        data-testid="filter_button"
        tabindex="0"
        type="button"
      >
        <span
          class="c2"
        >
          <svg
            aria-hidden="true"
            class=""
            fill="currentColor"
            height="24px"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class=""
              clip-rule="evenodd"
              d="M7 6h10l-5.01 6.3L7 6zm-2.75-.39C6.27 8.2 10 13 10 13v6c0 .55.45 1 1 1h2c.55 0 1-.45 1-1v-6s3.72-4.8 5.74-7.39A.998.998 0 0018.95 4H5.04c-.83 0-1.3.95-.79 1.61z"
              data-testid="eds-icon-path"
              fill-rule="evenodd"
              height="24"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
